atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/StartTransactionRequest.java

```diff
@@ -13,23 +13,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.palantir.atlasdb.transaction.api;
 
 import org.immutables.value.Value;
 
-import com.palantir.lock.v2.LockToken;
-
 @Value.Immutable
-public interface TransactionAndImmutableTsLock {
-
-    Transaction transaction();
-
-    LockToken immutableTsLock();
-
-    static TransactionAndImmutableTsLock of(Transaction transaction, LockToken immutableTsLock) {
-        return ImmutableTransactionAndImmutableTsLock.builder()
-                .transaction(transaction)
-                .immutableTsLock(immutableTsLock)
-                .build();
-    }
+public interface StartTransactionRequest {
+    PreCommitCondition preCommitCondition();
```

Comment:
Possibly, this dealing with List<PreCommitCondition> might cleanup some more code (where we have CombinedCondition* type things), but I'd hold off for now.